File: python geometry/magnet_geometry.py
from geant_classes import *
from globals import *

size_X = 16.0*mm
size_Y =  8.0*mm
size_Z = 16.0*mm
gap    = 16.0*mm


class MagnetGeometry(G4LogicalVolume):
    def __init__(self):
        magnet_solid = G4Box("magnet",
                             size_X/2,
                             size_Y/2,
                             size_Z/2)

        G4LogicalVolume.__init__(self, magnet_solid, G4Material("NdFeB"), "magnet")

File: source stephan/magnet_geometry.cc
 
#include "magnet_geometry.hh"

#include <G4Material.hh>
#include <G4Box.hh>
#include <G4PVPlacement.hh>
#include <G4VisAttributes.hh>
#include <G4Colour.hh>

const double magnet_geometry::size_X = 16*mm;
const double magnet_geometry::size_Y =  8*mm;
const double magnet_geometry::size_Z = 16*mm;

const double magnet_geometry::gap    = 16*mm;

magnet_geometry::~magnet_geometry()
{
}

magnet_geometry::magnet_geometry()

  : G4LogicalVolume(new G4Box("magnet",
			      size_X/2, size_Y/2, size_Z/2),
		    G4Material::GetMaterial("NdFeB"),
		    "magnet")

{
  G4VisAttributes *va = new G4VisAttributes(G4Colour(1, 0, 1));
  va->SetForceSolid(true);
  SetVisAttributes(va);
}
